jwst/pixel_replace/pixel_replace_step.py

```diff
@@ -81,9 +82,45 @@ def process(self, input):
                 'n_adjacent_cols': self.n_adjacent_cols,
             }
 
-            replacement = PixelReplacement(result, **pars)
-            replacement.replace()
-
-            self.record_step_status(replacement.output, 'pixel_replace', success=True)
-            return replacement.output
-
+            if isinstance(input_model, datamodels.ModelContainer):  # calspec3 case for IFU data
+                output_model = input_model.copy()
+                
+                # Setup output path naming if associations are involved.
+                asn_id = None
+                try:
+                    asn_id = self.input_model.meta.asn_table.asn_id
+                except (AttributeError, KeyError):
+                    pass
+                if asn_id is None:
+                    asn_id = self.search_attr('asn_id')
+                if asn_id is not None:
+                    _make_output_path = self.search_attr(
+                        '_make_output_path', parent_first=True
+                    )
+
+                    self._make_output_path = partial(
+                        _make_output_path,
+                        asn_id=asn_id
+                    )
+
+                for i, model in enumerate(input_model):
+                    if not isinstance(model, datamodels.IFUImageModel):
+                        self.log.error(f'Input is of type {str(type(input_model))} for which')
+                        self.log.error('pixel_replace does not have an algorithm.\n')
+                        self.log.error('Pixel replacement will be skipped.')
+                        output_model[i].meta.cal_step.pixel_replace = 'SKIPPED'
+                    else:
+                        self.log.debug('Input is an IFUImageModel.')
+                        replacement = PixelReplacement(model, **pars)
+                        replacement.replace()
+                        self.record_step_status(replacement.output, 'pixel_replace', success=True)
+                        output_model[i] = replacement.output
+                        o_path_name = self.make_output_path(basepath=output_model[i].meta.filename)
```

Comment:
I think something like
```suggestion
                        o_path_name = self.make_output_path(basepath=output_model[i].meta.filename)
                        self.output_file = o_path_name
```
will have the desired behavior. But should first check that self.output_file is None, so as to avoid overriding user specification of output_file.